### DISCUSSION TITLE
Turbine Turbulence Intensity values

### DISCUSSION THREAD

**Discussion by prith-gs** [NatLabRockies/floris#906] [General]
Hello,

I've a very simple setup with 2 Turbines and wind coming in from the west with wind speed = 9 m/s and TI = 6 %.
I'm using FLORIS v3.5 and the nrel_5MW power curve which comes with FLORIS.
This is the code I have.
![image](https://github.com/NREL/floris/assets/103917694/e4d48815-a02f-4ea4-a28f-7f92dbd2e1e2)

I'm using fi.get_turbine_TIs() to get the Turbulence Intensity at each turbine and I get the below array.
![image](https://github.com/NREL/floris/assets/103917694/9314102a-b807-41b4-aa80-8b2dbd326afa)

The first value is 6% (same as ambient) and the turbine in the wake has a higher TI as expected.

Now, When I add the crespo_hernandez model, I get a different value of TI for the Turbine in Wake.
![image](https://github.com/NREL/floris/assets/103917694/e93b5d85-ec81-4935-b693-29a03497d52a)
![image](https://github.com/NREL/floris/assets/103917694/ce77e4a0-123c-493a-acaf-38cda993a84a)

My questions are:
1. when I'm using fi.get_turbine_TIs() to get TI values at each turbine with the Turbulence model set to 'none'. How is the TI for turbine in Wake calculated? Is FLORIS using a different Turbulence model?
2. When I use the turbopark model with the Turbulence model set to 'none', I get the below array as an output where for both the turbines, the TI is 6%.
![image](https://github.com/NREL/floris/assets/103917694/57cebda4-820c-4764-bc3c-aee8ee523d6e)
 Why is it that when I use the jensen model with the Turbulence model set to 'none', I get a different TI for the Turbine in Wake? Ideally if I set the Turbulence model to 'none', there's no model used to calculate Turbulence variations inside the field, right? So, the output TI array should have the ambient TI for all the Turbines and for all wind speeds and directions.
 
 It would be great help if you can answer me this. I've been using FLORIS for over 2 years and I really appreciate all the updates and efforts put into developing this tool. Thanks! :)

**Comment by rafmudaf** [comment]
Hi @prith-gs

On the first point, this is a bug. It was a pointed out in #893 and fixed for v4 in #894. If you are not able to upgrade to the latest version, you could make the changes from that pull request in your version. If you do this, please consider submitting a pull request to the `v3` branch so that the bug fix is available for all users still on that version.

On the second question related to the TurbOPark model in FLORIS, I don't have a solid answer. This model was implemented with a [unique solver](https://github.com/NREL/floris/blob/main/floris/core/solver.py#L881). I'm wondering if it's related to [this comment](https://github.com/NREL/floris/blob/main/floris/core/solver.py#L1080) that implies that the wake area overlap calculation is invalid since the wake isn't computed at the current or downstream turbines, as in the other solvers. @bayc might have more information on this one.

Thanks,
Rafael

**Comment by prith-gs** [comment]
Thanks for the quick response. I'll upgrade to the latest version then.

One other thing that I noticed is that even when I use the 'crespo_hernandez' turbulence model with TurbOPark, I still get no added turbulence for turbines in wake. The output TI array is still 6 % for both turbines.
![image](https://github.com/NREL/floris/assets/103917694/289d3981-5f83-4a84-8ef3-c5a306beb370)

Please correct me if I'm wrong, this is because the TurbOPark model has no dependency on the choice of Turbulence and combination model. The TurbOPark model uses the Frandsen turbulence model and sum of squares superposition regardless of the user input. So, I'm guessing even with the unique solver (turbopark_solver) the local variations in TI are calculated but they are not exported. Is that right?

**Comment by misi9170** [comment]
Hi @prith-gs , sorry that it took me a while to get back to you here. I looked into this a bit, and I believe that the reason that the turbulence intensity is not changing for the downstream turbine using TurbOPark is has to do with [this TODO item](https://github.com/NREL/floris/blob/main/floris/core/solver.py#L1079). Essentially, the `area_overlap` computed in the `turbopark_solver` is all zeros, so the [subsequent lines that compute the wake added turbulence](https://github.com/NREL/floris/blob/main/floris/core/solver.py#L1091-L1097) also produce zeros.

I have opened an issue #909 to look at this; however, it may be some time before we address the issue, if we do at all. Note also that there is an open PR #907 that contains a different implementation of the TurbOPark model that runs through the `sequential_solver`, which may address the issue you have.
